lib/libc/minimal/include/time.h

```diff
@@ -51,6 +51,12 @@ typedef _SUSECONDS_T_ suseconds_t;
 struct tm *gmtime(const time_t *timep);
 struct tm *gmtime_r(const time_t *ZRESTRICT timep,
 		    struct tm *ZRESTRICT result);
+char *asctime(const struct tm *timeptr);
+char *asctime_r(const struct tm *ZRESTRICT tp, char *ZRESTRICT buf);
+struct tm *localtime(const time_t *timer);
+struct tm *localtime_r(const time_t *ZRESTRICT timer, struct tm *ZRESTRICT result);
+char *ctime(const time_t *clock);
+char *ctime_r(const time_t *clock, char *buf);
```

Comment:
@cfriedt do you think if I should guard this and above `asctime_r()` with `CONFIG_POSIX_THREAD_SAFE_FUNCTIONS`?

Reply:
I was planning on doing that for all of POSIX at some point, so sure, if you want to do that early.

I would suggest either of the following formats:

```
#if defined(CONFIG_POSIX_THREAD_SAFE_FUNCTIONS) || defined(__DOXYGEN__)
```

or

```
#if (_POSIX_THREAD_SAFE_FUNCTIONS >= 200809L) || defined(__DOXYGEN__)
```

Reply:
I changed the way the POSIX thread-safe functions are compiled, PTAL

Reply:
I noticed that Picolibc and Newlib do not have any conditions around the declarations of these functions

https://github.com/zephyrproject-rtos/picolibc/blob/764ef4e401a8f4c6a86ab723533841f072885a5b/newlib/libc/include/time.h#L123

It probably does make sense to declare the functions conditionally based on `_POSIX_THREAD_SAFE_FUNCTIONS >= 200809L` or some other revision. 

I would strongly advise against using per-function `CONFIG_` declarations.

Technically speaking, it looks like `_POSIX_THREAD_SAFE_FUNCTIONS` should be defined if `CONFIG_ POSIX_FILE_SYSTEM_R` and `CONFIG_POSIX_C_LANG_SUPPORT_R` is selected.

https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_subprofiles.html

In a follow-up PR, I'll likely be converting most of the `#ifdef CONFIG_POSIX_OPTION` conditions  to `#if defined(_POSIX_OPTION) || defined(__DOXYGEN__)`

Reply:
See the next comment.